### DISCUSSION TITLE
Converting type MX to Double

### DISCUSSION THREAD

**Discussion by agrace22** [casadi/casadi#3462] [Q&A]
Hi, is it possible to convert from SX to double in order to use logical comparisons? For example, Ncorr_CMP > min(compTable.NCorr) is giving errors since Ncorr_CMP uses an SX variable in its definition, while min(compTable.NCorr) is of type double. Is there a  way to extract the value in Ncorr_CMP and use it for a logical comparison or for other purposes?

This is the error:
Conversion to logical from casadi.SX is not possible.

Error in compressor_1stlaw (line 121)
if Ncorr_CMP > min(compTable.NCorr)

**Comment by jaeandersson** [comment]
An SX variable does not have a numeric value normally. A comparison operator like "Ncorr_CMP > min(compTable.NCorr)" will give a new expression that evaluates to either zero or one. So you can't use it in an "if".

Normally, logical comparisons have to be worked around by reformulating the model or by using support for conditionals (like Function.if_else).
